### ISSUE TITLE
Please read & provide the following

### ISSUE THREAD

**Issue by aa200647963** [facebookresearch/detr#491]
I find a problem with the model, In the detr.py loss_labels function, the background class is set 91, and the model works well, but when I set the background class to be 0, the model works badly, why? 
![image](https://user-images.githubusercontent.com/50197630/156707672-d2bb4b06-49f5-4079-b2ab-a78136fc0300.png)

**Comment by ZuyongWu** [comment]
91这个是空类，背景类是0

**Comment by idejie** [comment]
> 91这个是空类，背景类是0

Hi,

As the format of coco, the id 0 is non-sense or means background.

I'm confused about why drop the last dim for classification.

[detr/models/detr.py](https://github.com/facebookresearch/detr/blob/8a144f83a287f4d3fece4acdf073f387c5af387d/models/detr.py#L274-L275)

Lines 274 to 275 in [8a144f8](https://github.com/facebookresearch/detr/commit/8a144f83a287f4d3fece4acdf073f387c5af387d)

 prob = F.softmax(out_logits, -1) 
 scores, labels = prob[..., :-1].max(-1)

**Comment by altair199797** [comment]
@idejie 
the last last dim, or better the last element in the last dim is the "no object" class. During training if a bounding box is predicted that has no matching GT bounding box, the cross entropy loss assigns the "no object" class as target to that bounding box, meaning that the probability of all other class-categories is becoming less probable in the softmax distribution. During evaluation the postprocessing simply ignores that part of the class probability distribution, as it holds no information, only the actual class probabilities (of all other classes) are converted into scores.
However I don't understand why the "no object" class and the background class is not the same, since there is no bounding box in coco annotations that hold the class zero(background). Probably a programming hack to align with coco category assignments. Maybe someone knows more.
